src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -130,6 +130,7 @@ abstract class UnCurry extends InfoTransform
     def isByNameRef(tree: Tree) = (
          tree.isTerm
       && (tree.symbol ne null)
+      && !(tree.symbol.hasPackageFlag || tree.isInstanceOf[This] || tree.isInstanceOf[Super])
```

Comment:
Could we narrow it down to `tree.isInstanceOf[Ident]`?

Reply:
Nope, by-name types flow through generics, so can currently turn up in `Select`-s as well. It's a pretty murky area of the spec/implementation.

Reply:
👍 thanks!